infrastructure/src/main/kotlin/extension/RedmadrobotExtension.kt

```diff
@@ -119,3 +127,32 @@ public class TestOptions {
         useJunitPlatform = false
     }
 }
+
+public class DetektOptions {
+
+    /** Options for detektDiff task */
+    internal var detektDiffOptions: DetektDiffOptions? = null
+        private set
+
+    /** Enable Detekt checks only for modified files provided by git (compare with [branch]) */
+    public fun checkOnlyDiffWithBranch(
+        branch: String,
+        configure: DetektDiffOptions.() -> Unit = {},
+    ) {
+        require(branch.isNotBlank()) { "Base branch not provided" }
+
+        detektDiffOptions = DetektDiffOptions().apply {
+            configure()
+            baseBranch = branch
+        }
+    }
+}
+
+public class DetektDiffOptions {
+
+    /** Base branch to compare changes */
+    internal var baseBranch: String = ""
+
+    /** List of file extensions to check */
+    public var fileExtensions: Set<String> = setOf(".kt")
```

Comment:
Should we also check `.kts` by default?

Reply:
I was thinking about `.kts`, and I thought that Detekt has rules with excluded `.kts` files, but now I think that we should include `.kts` files to keep the behavior using the `detektAll` task, which also checks `.kts` files.
I will fix it now.

Reply:
Added.